File: backend/springboot/src/main/java/lx/iseau/feature/controltower/HeartRateRequest.java
// lx.iseau.feature.watch.HeartRateRequest.java
package lx.iseau.feature.controltower;

import lombok.Data;
import lombok.NoArgsConstructor;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class HeartRateRequest {

    private Integer userNumber;  // 워치 사용자의 고유 번호
    @JsonProperty("heart_rate")
    private Integer heartRate;   // 심박수 (BPM)
    @JsonProperty("occurred_at")
    private String occurredAt;   // ISO 8601 시간 문자열 (예: "2025-11-11T22:15:00Z")
    private Boolean isEmergency; // 긴급 알림 여부 (true/false)
}

File: ISeaUApp/app/src/main/java/com/lx/iseau/data/HeartRateRequest.kt
package com.lx.iseau.data

import kotlinx.serialization.Serializable

@Serializable
data class HeartRateRequest(
    //Int 타입을 Nullable(Int?)로 변경하여 Java Integer와 호환성 맞추기
    val userNumber: Int?,
    val heartRate: Int?,
    val occurredAt: String, // ISO 8601 형식
    val isEmergency: Boolean = false
)